packages/malloy-syntax-highlight/README.md

```diff
@@ -0,0 +1,120 @@
+
+## Generating a ground-truth tokenization artifact
+
+This package uses the term tokenization(s) to refer to a ground-truth artifact containing information about the scope names (ex. constant.numeric.date, keyword.control.select) and the color data (hex string color representation) applied to certain patterns in user-defined blocks of text. See `grammars/malloy/malloyTestInput` for an example of these user-defined blocks. Each inner array represents a block of lines to tokenize and highlight, and we would like these blocks to exhaust the constructs that our language supports through definitions that at least somewhat resemble valid user-written Malloy code. See `grammars/malloy/tokenizations/darkPlus.ts` for an example of a tokenization artifact. Each tokenization artifact also identifies the indices where new scope names or new colors begin in the test input.
+
+Since TextMate grammars are used to provide syntax highlighting for Malloy dialects in VSCode and Monarch grammars do the same for the Monaco editor, both of which are Microsoft products with builtin theme support, color data from these tests depends on VSCode themes to map many scope names to one color. While the notion of themes is far broader than what VSCode has championed, many code editors go out of their way to support custom theming and even pull in existing VSCode themes due to their ubiquity. This package allows us to run TextMate tests against the VSCode Dark Plus (Dark+) and Light Plus (Light+) themes, which are two of the richest, most ubiquitous themes offered by VSCode. Read more about sourcing these themes in `themes/THEMES.md`.
+
+Another motivator for tracking color data is ensuring parity between our two current grammar formats, which is impossible to do using scope names alone because TextMate applies many scope names to patterns while Monaco only applies one per pattern. The task of knowing which TextMate scope name will actually be selected to style text matching these patterns is determined entirely by the theme that the editor is using, which also provides access to color data. Besides,the ultimate goal is to ensure that users see the same colors across environments, so what better way to do this than by inspecting colors programmatically?
+
+To create a ground truth tokenization artifact, you can use `scripts/generateLanguageTokenizationFile.ts` and change the config object passed into the `generateTokenizationFile(...)` function. An example config object can be found at `test/config/textmate/malloyDarkPlusConfig.ts`. Note that we can only generate tokenization artifacts from TextMate grammars at the moment since most other syntax highlighters are downstream from these ones. To run an example of generating an artifact, use:
+
+```
+npm run gen-malloy-tokens
```

Comment:
Running this command makes a `darkPlus.ts` with a ton of formatting differences from what is currently checked in. 

Maybe we can tell the auto-formatter to ignore this file. Also, it might be good to have the generator add a line at the top "this file is generated, please do not edit manually".

Reply:
The auto-formatting is likely from the call to util.inspect which is the best solution I found for writing raw JS objects to output files that doesn't include quoted properties and string escapes like JSON.stringify. In the next PR, I'll look into how we can possibly run gts fix after writing to the output file.